doc/lsst.daf.butler/queries.rst

```diff
@@ -442,6 +452,8 @@ Few examples of valid expressions using some of the constructs:
 
     ingest_date < T'2020-11-06 21:10:00'
 
+    flat.dataset_id = UUID('e15ab039-bc8b-4135-87c5-90902a7c0b22')
```

Comment:
You need to specify the dataset type? We don't have any documentation on the `bind` form but if I have a list of arbitrary UUIDs and I want a query that retrieves them, do I do `where="dataset_id IN (ids)", bind={"ids": list_of_UUID}`? Where that is `list[uuid.UUID]`?

Reply:
If you have a query that only has one dataset type in it, you don't have to specify the dataset type.  The problem of querying for all datasets given UUIDs is tricky because they might have arbitrary dimensions, and half-completed effort to query for multiple dataset types in batches with the same dimensions helps with that but doesn't completely solve it.  We'd probably have do that kind of query in two phases: first look up all of the dataset types and run collections from the UUIDs, and then do one query for each dimension group.

Reply:
Ah. So none of this solves my SIAv2 problem where I can be given arbitrary IVOIDs and know nothing about where they come from. Sounds like I will have to do the loop over `Butler.get_dataset`. I was hoping I could do a WHERE of `dataset_id in (ids)` potentially with some more filtering clauses.

Reply:
Unfortunately correct.  We probably want to look into a dedicated vectorized version of that that could do its own query against the `dataset` table and then delegate to the rest of the query system to get `DatasetRefs`.